### PR TITLE
Dev databinding twoway

### PR DESCRIPTION
**Pull request by OyaCanli**
-Fixed the problem during rotation by creating a separate viewmodel. 
-Fixed double checked lockings in singletons. 
-Simplified repository. 
-Updated Gradle. 
-Changed attachCategories helper method
-Added readme and screenshots.

### CODE REVIEW

File: TwoWayDatabinding/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Two-Way Databinding
+This sample is designed to demonstrate two-way databinding. It's a simple inventory app, with the theme Toy Inventory. 
+It contains a single activity with two fragments: First fragment shows the list of toys, whereas second fragment is used for adding or editing toys.
+You can see the use of two-way databinding in AddToyFragment. We used edittext, checkboxes, radiobuttons and a spinner that all works two-way:
+It shows the chosen toy when you're editting a toy, but it also listens for changes. When you are entering a new entry, it is again listening for the changes simultaneously and keeps it in viewmodel. And all these without setting any listeners from java code.If you have previously done an inventory app without databinding, you'd be surprised how this is done with very few code.
```
Comment (rajtheinnovator) [inline TwoWayDatabinding/README.md:5]: @OyaCanli the README file looks all nice but is there another way to write it in even more simpler way? I mean, in a way where we can explain the working way of two-way-databinding as a brief but still the actual working than just giving the idea? Or should we leave that part for tutorial?

**Review by rajtheinnovator — COMMENTED** [review]
I've added one comment in the review here. However that's not critical but if we can simplify that even more that'd be so helpful. If you think that current description is suffice, we're good to go with merging the PR

**Comment by OyaCanli** [comment]
I didn't try to teach two-way databinding in readme. I thought we'll do it in the article. We can also put a link of it in all readmes. 
Or if you think it is better to teach within readmes as well, we can copy paste and adjust related parts of the article.

**Comment by rajtheinnovator** [comment]
@OyaCanli No. I didn't mean explaining it all in the README file itself. I was looking if more explanation is possible in less words. Let's keep the current structure and leave the detail description for now. At the end, we can insert the article link or if needed, we can create wikis for the sample.

**Pull request merged**